### PR TITLE
Add time to publication date and remove deprecated date fields 

### PR DESCRIPTION
**Pull request by Cloud11PL**
* Changes 'date' fields in product availability and publication to 'datetime'
* Changes 'date' fields in page availability and publication to 'datetime'
* Replaces deprecated fields with correct ones:
`availableForPurchase` -> `availableForPurchaseAt`
`publicationDate` -> `publishedAt`
`Date` -> `DateTime`

## What type of PR is this?
- [ ] 💅 Refactor
- [x] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Usage Instructions, Screenshots, Recordings

CleanShot 2024-05-10 at 11 01 19
CleanShot 2024-05-10 at 11 01 42
CleanShot 2024-05-10 at 11 02 05

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

**Comment by mirekm** [comment]
Do we still need to fill the old date fields (`availableForPurchase` and `publicationDate`) with the selected date?

**Review by Droniu — APPROVED** [review]

**Comment by Cloud11PL** [comment]
@mirekm No, Saleor syncs old fields with new fields 🙂

**Review by andrzejewsky — APPROVED** [review]

**Review by Droniu — DISMISSED** [review]

File: src/components/DateTimeTimezoneField.test.tsx
Comment (Droniu) [inline src/components/DateTimeTimezoneField.test.tsx:1]: tests should follow AAA format

File: src/components/VisibilityCard/VisibilityCard.tsx [hunk 1]
```diff
@@ -130,53 +137,80 @@ export const VisibilityCard: React.FC<VisibilityCardProps> = props => {
...
+            });
+          }}
+          display="flex"
+          flexDirection="column"
+          gap={3}
+        >
+          <RadioGroup.Item id={`isPublished-true`} value="true">
+            <Box display="flex" __alignItems="baseline" gap={2}>
+              <Text>{messages.visibleLabel}</Text>
+              {isPublished && publishedAt && Date.parse(publishedAt) < dateNow && (
```
Comment (Droniu) [inline src/components/VisibilityCard/VisibilityCard.tsx:165]: this condition is a little difficult to read, i'd extract it but it's a nitpick really - up to you

File: src/components/VisibilityCard/VisibilityCard.tsx [hunk 2]
```diff
@@ -193,11 +227,11 @@ export const VisibilityCard: React.FC<VisibilityCardProps> = props => {
               firstOptionLabel={
                 <>
                   <p className={classes.label}>{messages.availableLabel}</p>
-                  {isAvailable &&
-                    availableForPurchase &&
-                    Date.parse(availableForPurchase) < dateNow && (
+                  {isAvailableForPurchase &&
+                    availableForPurchaseAt &&
+                    Date.parse(availableForPurchaseAt) < dateNow && (
```
Comment (Droniu) [inline src/components/VisibilityCard/VisibilityCard.tsx:232]: same here

**Review by Droniu — DISMISSED** [review]

**Review by andrzejewsky — DISMISSED** [review]

**Review by Droniu — APPROVED** [review]

**Review by andrzejewsky — CHANGES_REQUESTED** [review]
When I click on the date it supposed to close the datepicker

**Pull request merged**